src/components/ToDoList/ToDoList.jsx

```diff
@@ -6,31 +6,81 @@ import { Button, InputGroup, FormControl } from 'react-bootstrap';
 class ToDoList extends React.PureComponent {
 
   state = {
+    inputEditMessage: '',
     inputTaskValue: '',
     tasks: []
   }
 
   componentDidMount() {
     axios.get(`http://127.0.0.1:3001/api/tasks`).then(res => {
-      this.setState({ tasks: res.data });
+      this.setState({
+        tasks: res.data.map(task => {
+          task.editMode = false;
+          return task
+        })
+      });
     })
   }
 
-  setTaskValue = e => {
+  handleSetTaskValue = e => {
     this.setState({ inputTaskValue: e.target.value });
   }
 
-  addTask = () => {
+  handleSetEditMessage = e => {
+    this.setState({ inputEditMessage: e.target.value });
+  }
+
+  handleSetEditMode = id => {
+    this.setState({
+      tasks: this.state.tasks.map(task => {
+        if (id === task.id) {
+          task.editMode = !task.editMode;
+
+          if (task.editMode) {
+            this.setState({ inputEditMessage: task.message });
+          } else {
+            this.setState({ inputEditMessage: '' });
```

Comment:
Never call `setState` in a loop. You call rerender with each `setState`.
And you've called `setState` inside `setState`.
You don't need `setState` in a loop at all here.